lib/roast/helpers/timeout_handler.rb

```diff
@@ -15,26 +15,35 @@ module Helpers
     #
     # @example With custom working directory
     #   output, status = TimeoutHandler.call("pwd", timeout: 10, working_directory: "/tmp")
+    #
+    # @example With custom grace period
+    #   output, status = TimeoutHandler.call("sleep 10", timeout: 10, grace_period: 1)
+    #
+    # @example Disable process group for TTY commands (like macOS security)
+    #   output, status = TimeoutHandler.call("security find-generic-password", timeout: 30, use_pgroup: false)
     class TimeoutHandler
       DEFAULT_TIMEOUT = 30
-      MAX_TIMEOUT = 300
+      DEFAULT_GRACE_PERIOD = 5
 
       class << self
         # Execute a command with timeout using Open3 with proper process cleanup
         # @param command [String] The command to execute
         # @param timeout [Integer] Timeout in seconds
         # @param working_directory [String] Directory to execute in (default: Dir.pwd)
+        # @param use_pgroup [Boolean] Whether to spawn process in its own process group (default: true)
         # @return [Array<String, Integer>] [output, exit_status]
         # @raise [Timeout::Error] When command exceeds timeout duration
-        def call(command, timeout: DEFAULT_TIMEOUT, working_directory: Dir.pwd)
+        def call(command, timeout: DEFAULT_TIMEOUT, working_directory: Dir.pwd, grace_period: DEFAULT_GRACE_PERIOD, use_pgroup: true)
           timeout = validate_timeout(timeout)
           output = ""
           exit_status = nil
           wait_thr = nil
 
           begin
             Timeout.timeout(timeout) do
```

Comment:
I found a [gist](https://gist.github.com/lpar/1032297) that replaces the `Timeout` by something cleaner. Seems worth trying :)